test/jdk/java/foreign/TestSegments.java

```diff
@@ -129,26 +129,35 @@ public void testSmallSegmentMin() {
     }
 
     @Test(dataProvider = "segmentFactories")
-    public void testAccessModesOfFactories(Supplier<SegmentFactory> segmentFactorySupplier) {
-        SegmentFactory segmentFactory = segmentFactorySupplier.get();
-        MemorySegment segment = segmentFactory.segment();
+    public void testAccessModesOfFactories(Supplier<MemorySegment> segmentSupplier) {
+        MemorySegment segment = segmentSupplier.get();
         assertFalse(segment.isReadOnly());
-        segmentFactory.tryClose();
+        tryClose(segment);
+    }
+
+    static void tryClose(MemorySegment segment) {
+        if (segment.session().isCloseable()) {
+            segment.session().close();
+        }
     }
 
     @DataProvider(name = "segmentFactories")
     public Object[][] segmentFactories() {
-        List<Supplier<SegmentFactory>> l = List.of(
-                () -> SegmentFactory.ofArray(() -> MemorySegment.ofArray(new byte[] { 0x00, 0x01, 0x02, 0x03 })),
-                () -> SegmentFactory.ofArray(() -> MemorySegment.ofArray(new char[] {'a', 'b', 'c', 'd' })),
-                () -> SegmentFactory.ofArray(() -> MemorySegment.ofArray(new double[] { 1d, 2d, 3d, 4d} )),
-                () -> SegmentFactory.ofArray(() -> MemorySegment.ofArray(new float[] { 1.0f, 2.0f, 3.0f, 4.0f })),
-                () -> SegmentFactory.ofArray(() -> MemorySegment.ofArray(new int[] { 1, 2, 3, 4 })),
-                () -> SegmentFactory.ofArray(() -> MemorySegment.ofArray(new long[] { 1l, 2l, 3l, 4l } )),
-                () -> SegmentFactory.ofArray(() -> MemorySegment.ofArray(new short[] { 1, 2, 3, 4 } )),
-                () -> SegmentFactory.ofImplicitSession(session -> MemorySegment.allocateNative(4, 8, session)),
-                () -> SegmentFactory.ofImplicitSession(session -> MemorySegment.allocateNative(JAVA_INT, session)),
-                () -> SegmentFactory.ofImplicitSession(session -> MemorySegment.allocateNative(4, session))
+        List<Supplier<MemorySegment>> l = List.of(
```

Comment:
You will see changes like these in tests - basically I've reverted the test to a simpler form (as per before the API change being reverted here).